### PR TITLE
Command-line arguments with no value that are Bool settings should be implicitly set to true.

### LINKED ISSUE #70291: If a command-line argument has no value, and it is a setting with Bool type, assume it has value true.
```
clickhouse-client --optimize_aggregation_in_order --query "SELECT ..."
```

### PR DESCRIPTION
**Pull request by davidtsuk**
Closes #70291 
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Command-line arguments for Bool settings are set to true when no value is provided for the argument (e.g. `clickhouse-client --optimize_aggregation_in_order --query "SELECT 1"`).

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: tests/queries/0_stateless/03250_client_default_bool_cli_settings_with_no_value_to_true.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+# Tags: no-fasttest
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+$CLICKHOUSE_CLIENT --optimize_aggregation_in_order --query "SELECT 1" 2>&1 | grep -q "1" && echo "OK" || echo "FAIL"
```
Comment (pufit) [inline tests/queries/0_stateless/03250_client_default_bool_cli_settings_with_no_value_to_true.sh:8]: Let's do `SELECT value FROM system.settings WHERE name = 'optimize_aggregation_in_order'` instead just selecting 1

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit c876b17702edaaa8683fae3e1f85b426b07a8f2b with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70459/c876b17702edaaa8683fae3e1f85b426b07a8f2b/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/70456/97b596123d5f704959b412581e3f2db4929c56df/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Review by pufit — APPROVED** [review]

File: src/Core/BaseSettingsProgramOptions.h [hunk 2]
```diff
@@ -11,8 +11,16 @@ template <typename T>
 void addProgramOptionAsMultitoken(T &cmd_settings, boost::program_options::options_description & options, std::string_view name, const typename T::SettingFieldRef & field)
 {
     auto on_program_option = boost::function1<void, const Strings &>([&cmd_settings, name](const Strings & values) { cmd_settings.set(name, values.back()); });
-    options.add(boost::shared_ptr<boost::program_options::option_description>(new boost::program_options::option_description(
-            name.data(), boost::program_options::value<Strings>()->multitoken()->composing()->notifier(on_program_option), field.getDescription())));
+    if (field.getTypeName() == "Bool")
+    {
```
Comment (rschu1ze) [inline src/Core/BaseSettingsProgramOptions.h:15]: Given that l. 16/17 and l. 21/22 are almost the same, a brief comment could have been useful.

File: tests/queries/0_stateless/03250_client_default_bool_cli_settings_with_no_value_to_true.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+# Tags: no-fasttest
```
Comment (rschu1ze) [inline tests/queries/0_stateless/03250_client_default_bool_cli_settings_with_no_value_to_true.sh:2]: Hmm, why is this marked `no-fasttest`?

**Pull request merged**